### PR TITLE
feat: added min/max cpu and memory for kubernetes via starlark

### PR DESCRIPTION
**Pull request by Peeeekay**
Here is the doc PR with relevant description: https://github.com/kurtosis-tech/kurtosis/pull/708

### CODE REVIEW

**Comment by Peeeekay** [comment]
Sample Starlark
If new fields and old fields are set, ignore the old fields:-
```
    postgres = plan.add_service(
        name = "postgres",
        config = ServiceConfig(
            image = "postgres:15.2-alpine",
            ports = {
                POSTGRES_PORT_ID: PortSpec(5432, application_protocol = "postgresql"),
            },
            env_vars = {
                "POSTGRES_DB": POSTGRES_DB,
                "POSTGRES_USER": POSTGRES_USER,
                "POSTGRES_PASSWORD": POSTGRES_PASSWORD,
            },
            files = {
                SEED_DATA_DIRPATH: data_package_module_result.files_artifact,
            },
            max_cpu=1200,
            cpu_allocation=1000,
            max_memory=2048,
            min_cpu=100,
            min_memory=1024,
        ),
    )

```

Result:
```
 resources:
        limits:
          cpu: 1200m
          memory: 2048M
        requests:
          cpu: 100m
          memory: 1024M
```

**Comment by Peeeekay** [comment]
Sample Starlark:
```
 postgres = plan.add_service(
        name = "postgres",
        config = ServiceConfig(
            image = "postgres:15.2-alpine",
            ports = {
                POSTGRES_PORT_ID: PortSpec(5432, application_protocol = "postgresql"),
            },
            env_vars = {
                "POSTGRES_DB": POSTGRES_DB,
                "POSTGRES_USER": POSTGRES_USER,
                "POSTGRES_PASSWORD": POSTGRES_PASSWORD,
            },
            files = {
                SEED_DATA_DIRPATH: data_package_module_result.files_artifact,
            },
            max_cpu=1200,
            min_cpu=100,
            max_memory=2048,
            min_memory=1024,
        ),
    )
```


```
resources:
        limits:
          cpu: 1200m
          memory: 2048M
        requests:
          cpu: 100m
          memory: 1024M
```

File: api/typescript/src/core/kurtosis_core_rpc_api_bindings/api_container_service_grpc_web_pb.js [hunk 1]
```diff
@@ -7,7 +7,7 @@
 // Code generated by protoc-gen-grpc-web. DO NOT EDIT.
 // versions:
 // 	protoc-gen-grpc-web v1.4.2
-// 	protoc              v3.19.1
+// 	protoc              v3.15.6
```
Comment (laurentluce) [inline api/typescript/src/core/kurtosis_core_rpc_api_bindings/api_container_service_grpc_web_pb.js]: Should we align on a version here?

File: api/golang/core/lib/services/service_config_builder.go [hunk 2]
```diff
@@ -23,10 +23,17 @@ type ServiceConfigBuilder struct {
 	cmdArgs                    []string
 	envVars                    map[string]string
 	filesArtifactMountDirpaths map[string]string
-	cpuAllocationMillicpus     uint64
-	memoryAllocationMegabytes  uint64
-	privateIPAddrPlaceholder   string
-	subnetwork                 string
+
+	// these two fields will be deprecated soon
```
Comment (laurentluce) [inline api/golang/core/lib/services/service_config_builder.go]: Should we remove this comment?

File: core/server/api_container/server/startosis_engine/kurtosis_starlark_framework/test_engine/service_config_full_framework_bc_test.go
Comment (laurentluce) [inline core/server/api_container/server/startosis_engine/kurtosis_starlark_framework/test_engine/service_config_full_framework_bc_test.go:1]: Can you spell out `bc`, I don't know what it means.

File: api/typescript/src/core/kurtosis_core_rpc_api_bindings/api_container_service_grpc_web_pb.js (hunk 1 above)
Comment (Peeeekay) [reply api/typescript/src/core/kurtosis_core_rpc_api_bindings/api_container_service_grpc_web_pb.js]: yeah; sure; probably the latest.

**Review by laurentluce — APPROVED** [review]

**Comment by h4ck3rk3y** [comment]
neat stuff :) Will use!

**Pull request merged**